internal/jobworker/job/service.go

```diff
@@ -0,0 +1,186 @@
+// Package job provides types for interacting with jobworker jobs, being
+// arbitrary commands running on the host system.
+package job
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/tjper/teleport/internal/jobworker/cgroup"
+	"github.com/tjper/teleport/internal/jobworker/output"
+	"github.com/tjper/teleport/internal/log"
+
+	"github.com/google/uuid"
+	"golang.org/x/sys/unix"
+)
+
+// logger is an object for logging package events to stdout.
+var logger = log.New(os.Stdout, "job")
+
+var (
+	// ErrServiceClosing indicates a StartJob call was made while the Service
+	// was closing down.
+	ErrServiceClosing = errors.New("service closing")
+
+	// ErrJobAlreadyStarted indicates Service attempted to start a Job that had
+	// already started. Highly unlikely for different Job instances as UUIDs are
+	// used as identifiers.
+	ErrJobAlreadyStarted = errors.New("job already started")
+
+	// ErrJobNotFound indicates the Job is not accessible through the Service.
+	ErrJobNotFound = errors.New("job not found")
+)
+
+// ICgroupService specifies Service interactions with cgroup.
+type ICgroupService interface {
+	CreateCgroup(...cgroup.CgroupOption) (*cgroup.Cgroup, error)
+	PlaceInCgroup(cgroup.Cgroup, int) error
+	RemoveCgroup(uuid.UUID) error
+}
+
+// NewService creates a new Service intance.
+func NewService(cgroups ICgroupService) (*Service, error) {
+	if err := os.MkdirAll(output.Root, output.FileMode); err != nil {
+		return nil, fmt.Errorf("mkdir job service output; path: %v, error: %w", output.Root, err)
+	}
+
+	return &Service{
+		mutex:   new(sync.RWMutex),
+		healthy: true,
+		jobs:    new(sync.Map),
+		cgroups: cgroups,
+	}, nil
+}
+
+// Service facilitates job interactions.
+type Service struct {
+	mutex *sync.RWMutex
+	// healthy indicates if Service is accepting to jobs to start.
+	healthy bool
+	// jobs is an mapping of Job.ID keys to *Job instances. The sync.Map type has
+	// been used because the data structure is always expanding and deletes never
+	// occur.
+	jobs    *sync.Map
+	cgroups ICgroupService
+}
+
+// StartJob starts the job.
+func (s *Service) StartJob(_ context.Context, job Job, options ...cgroup.CgroupOption) error {
```

Comment:
Use *Job so invocations better match with job.New()

```suggestion
func (s *Service) StartJob(_ context.Context, job *Job, options ...cgroup.CgroupOption) error {
```